packages/blade/src/components/Table/_decisions/decisions.md

```diff
@@ -0,0 +1,541 @@
+# Table API Decisions <!-- omit in toc -->
+
+A table component helps in displaying data in a grid format, through rows and columns of cells. Table facilitates data organisation and allow users to: scan, sort, compare, and take action on large amounts of data.
+
+<img src="./table-thumbnail.png" width="760" alt="Table thumbnail" />
+
+- [Features](#features)
+- [Out of scope](#out-of-scope)
+- [Anatomy](#anatomy)
+- [API](#api)
+  - [API Design Approaches](#api-design-approaches)
+    - [1. Composable API](#1-composable-api)
+      - [Pros](#pros)
+      - [Cons](#cons)
+    - [2. Compact API](#2-compact-api)
+      - [Pros](#pros-1)
+      - [Cons](#cons-1)
+    - [Comparison](#comparison)
+    - [Decision](#decision)
+      - [1. Only Composable API](#1-only-composable-api)
+      - [2. Only Compact API](#2-only-compact-api)
+      - [3. Both Composable \& Compact API](#3-both-composable--compact-api)
+      - [Current API Usage Across Razorpay Projects](#current-api-usage-across-razorpay-projects)
+      - [Blade Team's Recommendation](#blade-teams-recommendation)
+      - [Conclusion](#conclusion)
+  - [Final API](#final-api)
+- [Table Library Evaluation](#table-library-evaluation)
+  - [Why a library?](#why-a-library)
+  - [What are we looking for in a library?](#what-are-we-looking-for-in-a-library)
+  - [Libraries evaluated](#libraries-evaluated)
+    - [Heavy sized library - AG Grid](#heavy-sized-library---ag-grid)
+    - [Medium sized library - Tanstack's React Table](#medium-sized-library---tanstacks-react-table)
+    - [Small sized library - React Table Library](#small-sized-library---react-table-library)
+    - [Comparison](#comparison-1)
+
+
+## Features
+- Column Sorting
+- Row Selection - Single & Multiple
+- Pagination
+- Bulk Actions Toolbar
```

Comment:
We should allow toolbar for each row as well. I can see in an illustration below, row level toolbar is shown as its own separate `Actions` column at the end of all columns. This will not be an optimal design for horizontally scrolling tables as the `Actions` column will be hidden away at the very end.

Instead the row level toolbar can be shown as an hover element when cursor is on top of the row. This allows user to perform inline actions instantly.

<img width="815" alt="Screenshot 2023-10-30 at 11 55 09 AM" src="https://github.com/razorpay/blade/assets/95009160/498942de-65d9-41a6-87d4-71f2003ffcb3">

Reply:
The problem with such a design is that it hides crucial information like arrears and additions. This will be use-case specific. We can't make a generalized decision to hide the last two rows to show an action. Action can always have a place on the top toolbar or it can be a dedicated column that can be fixed on the right side.